src/modules/private-share-folder/private-sharing.usecase.ts

```diff
@@ -2,22 +2,30 @@ import { Injectable } from '@nestjs/common';
 import { Folder } from '../folder/folder.domain';
 import { User } from '../user/user.domain';
 import { SequelizePrivateSharingRepository } from './private-sharing.repository';
-import { SequelizeUserRepository } from '../user/user.repository';
-import { SequelizeFolderRepository } from '../folder/folder.repository';
 import { PrivateSharingFolder } from './private-sharing-folder.domain';
 import { PrivateSharingRole } from './private-sharing-role.domain';
+import { PrivateSharingFolderRolesRepository } from './private-sharing-folder-roles.repository';
 
 export class InvalidOwnerError extends Error {
   constructor() {
     super('You are not the owner of this folder');
   }
 }
+export class FolderNotSharedError extends Error {
+  constructor() {
+    super('This folder is not shared');
```

Comment:
You must add 

Object.setPrototypeOf(this, FolderNotSharedError .prototype)

this ensures that the created error instance has the correct prototype, and the proper functioning of instanceof in the controller's catch